src/devices/oregon_scientific_sl109h.c

```diff
@@ -0,0 +1,158 @@
+{
+    uint8_t calculated_checksum, actual_checksum;
+    uint8_t sum = 0;
+    int  actual_expected_comparison;
+
+    uint8_t out[CHECKSUM_BYTE_COUNT] = {0};
+    bitbuffer_extract_bytes(bitbuffer, row_index, CHECKSUM_START_POSITION, out, SL109H_MESSAGE_LENGTH - CHECKSUM_START_POSITION);
+
+    for(int i = 0; i < CHECKSUM_BYTE_COUNT; i++) sum += (out[i] & 0x0f) + (out[i] >> 4);
+
+    calculated_checksum = ((sum + channel) % 16);
+    actual_checksum     = bitbuffer->bb[row_index][0] >> 4;
+    actual_expected_comparison = (calculated_checksum == actual_checksum);
+
+    if(debug_output & !actual_expected_comparison) {
+        fprintf(stderr, "Checksum error in Oregon Scientific SL109H message.  Expected: %01x  Calculated: %01x\n", actual_checksum, calculated_checksum);
+        fprintf(stderr, "Message: ");
+        bitbuffer_print(bitbuffer);
+    }
+
+    return actual_expected_comparison;
+}
+
+
+static int oregon_scientific_callback_sl109h(bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    uint8_t *msg;
+    uint8_t channel_bits;
+
+    float temp_c;
+    uint8_t humidity;
+    int channel;
+    uint8_t id;
+    int status;
+    char time_str[LOCAL_TIME_BUFLEN];
+
+    for(int row_index = 0; row_index < bitbuffer->num_rows; row_index++) {
+        msg = bitbuffer->bb[row_index];
+
+        channel_bits = get_channel_bits(msg[0]);
+
+        if( !((bitbuffer->bits_per_row[row_index] == SL109H_MESSAGE_LENGTH)
+              && calculate_checksum(bitbuffer, row_index, channel_bits)) ) continue;
+
+        local_time_str(0, time_str);
+
+        temp_c = calculate_centigrade_decidegrees(bitbuffer, row_index) / 10.0;
+
+        humidity = calculate_humidity(bitbuffer, row_index);
+
+        id = get_id(bitbuffer, row_index);
+
+        channel = calculate_channel(channel_bits);
+
+        status = get_status(msg[3]);
+
+        data = data_make("time",          "",           DATA_STRING,                         time_str,
+                         "temperature_C", "Celcius",	DATA_FORMAT, "%.02f C", DATA_DOUBLE, temp_c,
+                         "humidity",      "Humidity",	DATA_FORMAT, "%u %%",   DATA_INT,    humidity,
+                         "channel",       "Channel",                            DATA_INT,    channel,
+                         "id",            "Id",                                 DATA_INT,    id,
+                         "status",        "Status",                             DATA_INT,    status,
+                         NULL);
+        data_acquired_handler(data);
+    }
+
+    return bitbuffer->num_rows;
+}
+
+static char *output_fields[] = {
+    "time",
+    "id",
+    "channel",
+    "status",
+    "temperature_C",
+    "humidity",
+    NULL
+};
+
+r_device oregon_scientific_sl109h = {
+    .name           = "Oregon Scientific SL109H Remote Thermal Hygro Sensor",
+    .modulation     = OOK_PULSE_PPM_RAW,
+    .short_limit    = 2800/*760*/,
+    .long_limit     = 4400/*1050*/,
```

Comment:
I just ran into that as well.  Looking at rtl_433.c it shows the values are divided by ((float) 1000000 / (float) samp_rate.  Where the sample_rate defaults to DEFAULT_SAMPLE_RATE which is 250000.  So it basically divides the values by 4.  I found a commit for this back in January that says "Switch protocol decoder limits to µs instead of "samples at 250000Hz"".  I wasted a few hours working on a decoder to find out the values were divided by 4.  I ended up just multiplying the values by 4 and moved on.  They should probably change all output of samples to µs.